### DISCUSSION TITLE
(beginner) Triying to change a message's text, but doesn't show up in game

### DISCUSSION THREAD

**Discussion by kz-n** [Minestom/Minestom#1236] [Help] [answered]
```java
public static void main(String[] args) {
      // Initialization
      MinecraftServer minecraftServer = MinecraftServer.init();
      InstanceManager instanceManager = MinecraftServer.getInstanceManager();

      DimensionType fullbright = DimensionType.builder(NamespaceID.from("minestom:full_bright"))
              .ambientLight(100.0f)
              .build();
      MinecraftServer.getDimensionTypeManager().addDimension(fullbright);

      // Create the instance
      InstanceContainer instanceContainer = instanceManager.createInstanceContainer(fullbright);
      // Set the ChunkGenerator
      instanceContainer.setGenerator(unit ->
              unit.modifier().fillHeight(0, 40, Block.OAK_LEAVES));

      // Add an event callback to specify the spawning instance (and the spawn position)
      GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
      globalEventHandler.addListener(PlayerLoginEvent.class, event -> {
          final Player player = event.getPlayer();
          event.setSpawningInstance(instanceContainer);
          player.setRespawnPoint(new Pos(0, 42, 0));
      });
      var handler = MinecraftServer.getGlobalEventHandler();
      handler.addListener(PlayerChatEvent.class, event -> {
          System.out.println("Before changes " + event.getMessage());
          event.setMessage("Test");
          System.out.println("After changes " + event.getMessage());
      });

      // Start the server on port 25565
      minecraftServer.start("0.0.0.0", 25565);
}
```

```
Before changes Test message
After changes Test
```

The output would suggest it actually changed the message, but I don't see this in game
![immagine](https://user-images.githubusercontent.com/64893077/178945250-42181009-ff5b-49f8-abfd-7398fce38c2e.png)

I'm using version `366f08f7f9`

**Comment by VendoAU** [comment]
Are you reading chat from the same account that is sending the message?
I am pretty sure it's not possible to change a message for the sender.

**Reply by kz-n** [reply]
Other users dont get their chat message changed either, Although I think I found an alternative to this, which is cancelling the chat event, and sending a message to all players online with my own chat styling, Not sure if its a good practice though.

**Reply by emortaldev** [reply]
cancelling and resending is likely the best solution, unless changing the format works

**Reply by FluxCapacitor2** [reply]
It seems like changing the `message` does not affect anything, but (as emortal said) changing the chat format does modify the message:
https://github.com/Minestom/Minestom/blob/366f08f7f930514aadae0c1a92a78322f7f01530/src/main/java/net/minestom/server/listener/ChatMessageListener.java#L53-L72
This means that you can replace your listener with code similar to this:
```java
var handler = MinecraftServer.getGlobalEventHandler();
handler.addListener(PlayerChatEvent.class, event -> {
    event.setChatFormat((event) -> Component.text("my chat message"));
});
```
... and you can use the PlayerChatEvent in that function, so you can see the player's original message.

**Reply by kz-n** [reply]
alright thanks

**Comment by kz-n** [comment, marked as answer]
From @FluxCapacitor2 (reposting because cannot set comment as an answer)
> It seems like changing the `message` does not affect anything, but (as emortal said) changing the chat format does modify the message:
> 
> https://github.com/Minestom/Minestom/blob/366f08f7f930514aadae0c1a92a78322f7f01530/src/main/java/net/minestom/server/listener/ChatMessageListener.java#L53-L72
> 
> This means that you can replace your listener with code similar to this:
> 
> ```java
> var handler = MinecraftServer.getGlobalEventHandler();
> handler.addListener(PlayerChatEvent.class, event -> {
>     event.setChatFormat((event) -> Component.text("my chat message"));
> });
> ```
> 
> ... and you can use the PlayerChatEvent in that function, so you can see the player's original message.
